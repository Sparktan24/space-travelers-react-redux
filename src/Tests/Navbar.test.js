import React from 'react';
import {
  render, screen, act, fireEvent,
} from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect'; 
import Navbar from '../Components/NavBar';

describe('Navbar functionality', () => {
  it('render correctly', () => {
    const { container } = render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );
    expect(screen.getByText('Missions')).toBeInTheDocument();
    expect(screen.getByText('Rockets')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  })
  it('navigates to the Missions page when Missions link is clicked', () => {
    const { container } = render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );
    act(() => {
      const missionsLink = screen.getByText('Missions');
      fireEvent.click(missionsLink);
    });
    expect(window.location.pathname).toBe('/Missions');
    expect(container.firstChild).toMatchSnapshot();
  });
})