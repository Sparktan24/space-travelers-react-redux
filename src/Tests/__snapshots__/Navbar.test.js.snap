// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar functionality navigates to the Missions page when Missions link is clicked 1`] = `
<nav
  class="nav-container"
>
  <div
    class="logo-container"
  >
    <img
      alt="logo"
      class="logo"
      src="planet.png"
    />
    Space Travelers' Hub
  </div>
  <div
    class="links-container"
  >
    <a
      class="link"
      href="/Rockets"
    >
      Rockets
    </a>
    <a
      aria-current="page"
      class="link-active line"
      href="/Missions"
    >
      Missions
    </a>
    <a
      class="link"
      href="/MyProfile"
    >
      My Profile
    </a>
  </div>
</nav>
`;

exports[`Navbar functionality render correctly 1`] = `
<nav
  class="nav-container"
>
  <div
    class="logo-container"
  >
    <img
      alt="logo"
      class="logo"
      src="planet.png"
    />
    Space Travelers' Hub
  </div>
  <div
    class="links-container"
  >
    <a
      class="link"
      href="/Rockets"
    >
      Rockets
    </a>
    <a
      class="link line"
      href="/Missions"
    >
      Missions
    </a>
    <a
      class="link"
      href="/MyProfile"
    >
      My Profile
    </a>
  </div>
</nav>
`;
